node/actors/bft/src/replica/leader_prepare.rs

```diff
@@ -105,6 +105,12 @@ pub(crate) enum Error {
     Internal(#[from] anyhow::Error),
 }
 
+impl PartialEq for Error {
+    fn eq(&self, other: &Self) -> bool {
+        self.to_string() == other.to_string()
+    }
+}
```

Comment:
That's a bad idea for the same reasons as for `bn254::Error`. This error type isn't public; OTOH, in this case, unlike for `bn254::Error`, the implementation itself is rather dubious and brittle w.r.t. to refactoring.

Reply:
https://github.com/matter-labs/era-consensus/pull/39#discussion_r1401072512